### DISCUSSION TITLE
Consider adding a licence and issue template

### DISCUSSION THREAD

**Discussion by TakshakRamteke** [gleam-lang/gleam#1368] [Questions & support]
I just recently came across this cool project and saw some really good issues to begin with... but after taking a look at the community section in the insights tab i saw that there is no License and Issue template for this project, 

so i just wanted to ask if the maintainers want to add those if yes then please let me know i would love to do it for you

**Comment by lpil** [comment]
Hi there!

Currently we have licence files in each project

- https://github.com/gleam-lang/gleam/blob/main/compiler-cli/LICENCE
- https://github.com/gleam-lang/gleam/blob/main/compiler-core/LICENCE
- https://github.com/gleam-lang/gleam/blob/main/compiler-wasm/LICENCE

But I think you've a point- it is not obvious where they are. It would be great to have one at the top level too.

An issue template could be good, though I would want to be careful to avoid any formalities discouraging people from opening issues. What might we want to include in a template?

**Reply by TakshakRamteke** [reply]
**We may include things like asking them if its a feature request, a question, a bug report, or other** 
so when you assign labels to them it would be easier for you to add them and if we want to we can use conventional commits style 

e.g: in case of feature requests the issue can be like `Feat : please consider adding this new feature`, or in case of a bug report we can make it like `Bug : this is a bug that i keep facing is there any solution maybe ?` something like this and we can have a pr template similar to this so its easy to distinguish what pr is for what

also we can add a line in the readme that says : the [compiler-cli](https://github.com/gleam-lang/gleam/blob/main/compiler-cli/LICENCE) , [compiler-core](https://github.com/gleam-lang/gleam/blob/main/compiler-core/LICENCE) and [compiler-wasm](https://github.com/gleam-lang/gleam/blob/main/compiler-wasm/LICENCE) for this project are individually licensed under Apache License

what do you think

**Reply by lpil** [reply]
I don't currently have any problems with assigning labels and such, mostly because we don't have any issues being opened. I don't think that would help us right now, maybe in a year or two.

How about we add a licence file to the top level? We could also remove the three nested ones.

**Reply by TakshakRamteke** [reply]
yeah we can definitely do that having one license at top level would cover every everything in the repo. ig, right ?

**Reply by lpil** [reply]
Yes, I think so. It would probably be the clearest like that

**Reply by TakshakRamteke** [reply]
Yes, if you like may i do that for you

**Reply by lpil** [reply]
Yes please, that would be great!
